### PR TITLE
Avoid 10 minute goroutine leak in error case for handled errors

### PR DESCRIPTION
**Pull request by liggitt**
Addresses https://github.com/moby/spdystream/issues/92 for handled errors

I think there's more we could do here to shorten the 10 minute hang for unhandled errors, but this is a smaller safer first improvement.

cc @aojea @seans3

### CODE REVIEW

**Comment by thaJeztah** [comment]
@dmcgowan @dims ptal

**Comment by thaJeztah** [comment]
FWIW; I'm also happy to add more maintainers from the k8s folks familiar with the code and how it's used in k8s if that helps; I think k8s is currently the only consumer of this module (leaving that to @dims and others to consider; just wanted to mention that I wouldn't have objections)

File: connection.go [hunk 1]
```diff
@@ -740,16 +740,14 @@ func (s *Connection) shutdown(closeTimeout time.Duration) {
 
 	if err != nil {
 		duration := 10 * time.Minute
-		time.AfterFunc(duration, func() {
```
Comment (aojea) [inline connection.go:743]: this avoids that if `s.shutdownChan <- err` is drained, we don't have a goroutine leaked during 10 minutes

https://go.dev/play/p/3PcdQYbcZdu

**Comment by aojea** [comment]
we are consuming it and it seems the plan we had to deprecate is going to take some  time, we are interested in keeping it solid and only to fix bugs

It should not be problematic, at least from my side, as it seems it has been running in production for lot of years

**Comment by aojea** [comment]
LGTM

**Review by dims — APPROVED** [review]
LGTM

File: connection.go (hunk 1 above)
Comment (dims) [reply connection.go:743]: ✅

**Pull request merged**